pkg/upgrademonitor/upgrade_monitor.go

```diff
@@ -194,6 +194,12 @@ func GenerateAndApplyMachineConfigNodes(parentCondition, childCondition *Conditi
 			pool = "master"
 		}
 
+		newMCNode.Spec.ConfigVersion = mcfgalphav1.MachineConfigNodeSpecMachineConfigVersion{
+			Desired: node.Annotations["machineconfiguration.openshift.io/desiredConfig"],
+		}
+		if newMCNode.Spec.ConfigVersion.Desired == "" {
+			newMCNode.Spec.ConfigVersion.Desired = "NotYetSet"
```

Comment:
Wondering if we should just stick to "" to reflect node annotation value or we are adding this string for better readability for user?

Reply:
the API type has to be len > 0. I think we should change this in 4.16 but the API team insisted on it. So, we are stuck with having some default value.

Reply:
ah, thanks for the context